pftools/python/parflow/tools/pf_backend.py

```diff
@@ -0,0 +1,562 @@
+import contextlib
+import dask
+import json
+import numpy as np
+import os
+import warnings
+import xarray as xr
+import yaml
+
+from pprint import pprint
+from . import util
+from .io import ParflowBinaryReader, read_stack_of_pfbs, read_pfb
+from collections.abc import Iterable
+from dask import delayed
+#from parflow.tools import hydrology
+#from parflowio.pyParflowio import PFData
+from typing import Mapping, List, Union
+from xarray.backends  import BackendEntrypoint, BackendArray, CachingFileManager
+from xarray.backends.locks import SerializableLock
+from xarray.core import indexing
+
+PARFLOW_LOCK = SerializableLock()
+NO_LOCK = contextlib.nullcontext()
+
+class ParflowBackendEntrypoint(BackendEntrypoint):
+
+    open_dataset_parameters = [
+            "filename_or_obj",
+            "drop_variables",
+            "name",
+            "meta_yaml",
+            "read_inputs",
+            "read_outputs",
+            "parallel",
+            "inferred_dims",
+            "inferred_shape"
+    ]
+
+    def open_dataset(
```

Comment:
Many undocumented methods in this class.   It is not very clear what some of the options do.   e.g. "read_inputs" and "read_outputs" ???

Reply:
Documented these.